src/components/card.js

```diff
@@ -1,32 +1,36 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
     this._data = {
       name: data.name,
       link: data.link,
+      id: data._id,
+      isLiked: data.isLiked,
     };
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
     this._cardSelector = cardSelector;
   }
 
   _setEventListeners() {
-    this._likeIcon.addEventListener("click", this._handleLikeIcon);
-
-    this._deleteBtn.addEventListener("click", this._handleDeleteCard);
-
+    this._likeIcon.addEventListener("click", () => {
+      this._handleLikeClick(this._likeIcon, this._data.isLiked, this._data.id);
+      this._data.isLiked = !this._data.isLiked;
```

Comment:
Needs correction you should update the likes state only after the successful api response

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```